smart-hadoop-support/smart-hadoop/src/main/java/org/smartdata/hdfs/metric/fetcher/InotifyEventFetcher.java

```diff
@@ -64,41 +68,92 @@ public InotifyEventFetcher(DFSClient client, MetaStore metaStore,
       ScheduledExecutorService service, InotifyEventApplier applier, Callable callBack) {
     this.client = client;
     this.applier = applier;
+    this.metaStore = metaStore;
     this.scheduledExecutorService = service;
-    this.fetchFinishedCallback = callBack;
+    this.finishedCallback = callBack;
     this.nameSpaceFetcher = new NamespaceFetcher(client, metaStore, service);
   }
 
   public void start() throws IOException {
+    Long lastTxid = getLastTxid();
+    if (lastTxid != null && lastTxid != -1 && canContinueFromLastTxid(client, lastTxid)) {
+      startFromLastTxid(lastTxid);
+    } else {
+      startWithFetchingNameSpace();
+    }
+  }
+
+  @VisibleForTesting
+  static boolean canContinueFromLastTxid(DFSClient client, Long lastId) {
+    try {
+      if (client.getNamenode().getCurrentEditLogTxid() == lastId) {
+        return true;
+      }
+      DFSInotifyEventInputStream is = client.getInotifyEventStream(lastId);
+      EventBatch eventBatch = is.poll();
+      return eventBatch != null;
+    } catch (Exception e) {
+      e.printStackTrace();
```

Comment:
I don't know whether "e.printStackTrace()" can output the info into log file or not, but maybe better to use LOG instead as it provides more flexibility and looks more formal. Thanks!